### PR TITLE
PolyhedralGeometry: Compute triangulations of polytope

### PR DESCRIPTION
**Pull request by lkastner**

### CODE REVIEW

**Comment by HereAround** [comment]
@lkastner: What is the status of this draft?

File: src/PolyhedralGeometry/triangulations.jl [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+ [1, 1]
+
+julia> all_triangulations(point_matrix(V))
+2-element Vector{Vector{Vector{Int64}}}:
+ [[1, 2, 3], [2, 3, 4]]
+ [[1, 2, 4], [1, 3, 4]]
+```
+"""
+function all_triangulations(pts::AnyVecOrMat)
+    input = homogenize(pts)
```
Comment (benlorenz) [inline src/PolyhedralGeometry/triangulations.jl]: Do we really want to allow a single vector as input?
We could use this instead, which allows passing `vertices(...)` directly:
```suggestion
function all_triangulations(pts::Union{SubObjectIterator{PointVector}, AbstractMatrix, Oscar.MatElem})
    input = homogenized_matrix(pts, 1)
```

We could define some aliases for these unions (for vertexlist, raylist and maybe more?) and unify our signatures a little bit.

File: src/PolyhedralGeometry/triangulations.jl [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+julia> c = cube(2,0,1)
+A polyhedron in ambient dimension 2
+
+julia> all_triangulations(c)
+2-element Vector{Vector{Vector{Int64}}}:
+ [[1, 2, 3], [2, 3, 4]]
+ [[1, 2, 4], [1, 3, 4]]
+```
+"""
+all_triangulations(P::Polyhedron) = all_triangulations(point_matrix(vertices(P)))
```
Comment (benlorenz) [inline src/PolyhedralGeometry/triangulations.jl]:
```suggestion
all_triangulations(P::Polyhedron) = all_triangulations(vertices(P))
```
This should work together with the previous change.

File: src/PolyhedralGeometry/triangulations.jl [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+A polyhedron in ambient dimension 2
+
+julia> V = vertices(c)
+4-element SubObjectIterator{PointVector{Polymake.Rational}}:
+ [0, 0]
+ [1, 0]
+ [0, 1]
+ [1, 1]
+
+julia> all_triangulations(point_matrix(V))
```
Comment (benlorenz) [inline src/PolyhedralGeometry/triangulations.jl]:
```suggestion
julia> all_triangulations(V)
```

File: src/PolyhedralGeometry/triangulations.jl [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+A polyhedron in ambient dimension 2
+
+julia> V = vertices(c)
+4-element SubObjectIterator{PointVector{Polymake.Rational}}:
+ [0, 0]
+ [1, 0]
+ [0, 1]
+ [1, 1]
+
+julia> regular_triangulations(point_matrix(V))
```
Comment (benlorenz) [inline src/PolyhedralGeometry/triangulations.jl]:
```suggestion
julia> regular_triangulations(V)
```

File: src/PolyhedralGeometry/triangulations.jl [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+2-element Vector{Vector{Vector{Int64}}}:
+ [[1, 2, 3], [2, 3, 4]]
+ [[1, 3, 4], [1, 2, 4]]
+```
+"""
+function regular_triangulations(pts::AnyVecOrMat)
+    input = homogenize(pts)
```
Comment (benlorenz) [inline src/PolyhedralGeometry/triangulations.jl]:
```suggestion
function regular_triangulations(pts::Union{SubObjectIterator{PointVector}, AbstractMatrix, Oscar.MatElem})
    input = homogenized_matrix(pts, 1)
```

File: src/PolyhedralGeometry/triangulations.jl [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+julia> c = cube(2,0,1)
+A polyhedron in ambient dimension 2
+
+julia> regular_triangulations(c)
+2-element Vector{Vector{Vector{Int64}}}:
+ [[1, 2, 3], [2, 3, 4]]
+ [[1, 3, 4], [1, 2, 4]]
+```
+"""
+regular_triangulations(P::Polyhedron) = regular_triangulations(point_matrix(vertices(P)))
```
Comment (benlorenz) [inline src/PolyhedralGeometry/triangulations.jl]:
```suggestion
regular_triangulations(P::Polyhedron) = regular_triangulations(vertices(P))
```

File: docs/src/PolyhedralGeometry/Polyhedra/auxiliary.md [hunk 7]
```diff
@@ -62,6 +63,7 @@ polarize(P::Polyhedron)
 project_full(P::Polyhedron)
 print_constraints(A::AnyVecOrMat, b::AbstractVector; trivial::Bool = false)
 print_constraints(P::Polyhedron; trivial::Bool = false)
+regular_triangulations
 solve_ineq(A::fmpz_mat, b::fmpz_mat)
```
Comment (benlorenz) [inline docs/src/PolyhedralGeometry/Polyhedra/auxiliary.md]:
```suggestion
regular_triangulations
secondary_polytope
solve_ineq(A::fmpz_mat, b::fmpz_mat)
```
Secondary polytope seems to be missing in the docs?

**Review by benlorenz — APPROVED** [review]

**Pull request merged**